### PR TITLE
fix: prevent error when minimap squashed

### PR DESCRIPTION
**Pull request by aminya**
Fix error when the middle pane is 0 width

minimap-width

Fixes https://github.com/atom-ide-community/minimap-plus/issues/15
Fixes https://github.com/atom-minimap/minimap/issues/702

Originally posted at https://github.com/atom-ide-community/minimap-plus/pull/13#issue-465109156 by @UziTech

### CODE REVIEW

**Review by aminya — COMMENTED** [review]
This does not fix the issue yet.

File: lib/canvas-layer.js [hunk 1]
```diff
@@ -61,11 +61,15 @@ module.exports = class CanvasLayer {
...
   }
 
   copyFromOffscreen () {
-    this.context.drawImage(this.offscreenCanvas, 0, 0)
+    if (this.offscreenCanvas.width > 0 && this.offscreenCanvas.height > 0) {
+      this.context.drawImage(this.offscreenCanvas, 0, 0)
+    }
   }
 
   copyPartFromOffscreen (srcY, destY, height) {
```
Comment (UziTech) [inline lib/canvas-layer.js:75]: looks like this method needs the check as well

**Review by aminya — COMMENTED** [review]
Are these `canvas.width` and `offscreenCanvas.width` values calculated on the fly or they are cached/constant?

**Comment by UziTech** [comment]
I believe they are calculated

**Comment by aminya** [comment]
Could you benchmark one call to them?

```
let t1 = window.performance.now()

// call to these

console.log(window.performance.now() - t1)
```

**Comment by UziTech** [comment]
I got `.07` with the if and `.09` without lol so it must be constant

**Comment by aminya** [comment]
Sounds good. I will test this tomorrow if I find some time.

**Pull request merged**